recipes/license_server.rb

```diff
@@ -25,19 +23,28 @@
 data_bag_item = bag.to_hash
 total_available_license_quota = 0
 
-license_groups = data_bag_item.inject('enterprise' => {}) do |hash, (key, value)|
+license_groups = data_bag_item.inject({}) do |hash, (key, value)|
   unless %w[id chef_type data_bag].include? key
     doc = Nokogiri::XML value
+    sourcetypes = doc.search('//sourcetype').map(&:text).join
     type = doc.at_xpath('/license/payload/type/text()').to_s
+    type = "#{type}_#{Digest::SHA256.hexdigest(sourcetypes).upcase}" if type == 'fixed-sourcetype'
     quota = doc.at_xpath('/license/payload/quota/text()').to_s.to_i
     expiration_time = doc.at_xpath('/license/payload/expiration_time/text()').to_s.to_i
-    total_available_license_quota += quota if type == 'enterprise' && expiration_time > Time.now.to_i
+    total_available_license_quota += quota if (type == 'enterprise' || type == "fixed-sourcetype_#{Digest::SHA256.hexdigest(sourcetypes).upcase}") && expiration_time > Time.now.to_i
     hash[type] ||= {}
+    unless hash[type].empty?
```

Comment:
this isn't currently working as intended.  This should work:
```
    if hash.length > 1
      fail 'Multiple license types are not currently supported'
    end
```

Reply:
no, the hash[type] ||= {} is sufficient to add an entry to the hash.

Reply:
That is true. Didn't see that.